include/NeoFOAM/fields/boundaryFields.hpp

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+#pragma once
+
+#include <Kokkos_Core.hpp>
+#include <iostream>
+#include "NeoFOAM/primitives/label.hpp"
+#include "NeoFOAM/primitives/scalar.hpp"
+#include "Field.hpp"
+
+#include "NeoFOAM/core/executor/executor.hpp"
+
+namespace NeoFOAM
+{
+// enum class boundaryType used in a kokkos view to categorize boundary
+// types
+
+
+template<typename T>
+class BoundaryFields
+{
+    /**
+     * @class boundaryFields
+     * @brief Represents the boundary fields for a computational domain.
+     *
+     * The boundaryFields class stores the boundary conditions and related information for a computational domain.
+     * It provides access to the computed values, reference values, value fractions, reference gradients, boundary types,
+     * offsets, and the number of boundaries and boundary faces.
+     */
```

Comment:
Docstring before class
```suggestion
/**
* @class boundaryFields
* @brief Represents the boundary fields for a computational domain.
*
* The boundaryFields class stores the boundary conditions and related information for a computational domain.
* It provides access to the computed values, reference values, value fractions, reference gradients, boundary types,
* offsets, and the number of boundaries and boundary faces.
*/
template<typename T>
class BoundaryFields
{
```